src/common/utils.cpp

```diff
@@ -832,58 +832,15 @@ void rtrim(std::string& s)
 // Returns true if the given str matches the given pattern.
 // Wildcards can be used in the pattern to match "any character"
 // e.g: %anto% matches Shantotto or Canto-Ranto
-// Modification of https://www.geeksforgeeks.org/wildcard-character-matching/
 bool matches(std::string const& target, std::string const& pattern, std::string const& wildcard)
 {
-    auto matchesRecur = [&](const char* target, const char* pattern, const char* wildcard, auto&& matchesRecur)
+    // Replace our wildcard with ".*", perform standard regex match.
```

Comment:
I'm in for the use of regex, but I think I'd prefer it if we used it instead of a nested replacement to swap it for the very standardized and understood `.*`. If people can't figure out .*, they shouldn't be on a computer.

Reply:
Updated to pure regex 
![image](https://github.com/LandSandBoat/server/assets/60417494/54e823f0-a48c-45c9-9637-0db1063f6faf)